Backend/src/test/java/ca/mcgill/cooperator/controller/CSVParserControllerIT.java

```diff
@@ -0,0 +1,487 @@
+
+        CourseOfferingDto courseOfferingDto = new CourseOfferingDto();
+        courseOfferingDto.setYear(2020);
+        courseOfferingDto.setSeason(Season.SUMMER);
+        courseOfferingDto.setCourse(courseDto);
+
+        mvcResult =
+                mvc.perform(
+                                post("/course-offerings")
+                                        .contentType(MediaType.APPLICATION_JSON)
+                                        .content(objectMapper.writeValueAsString(courseOfferingDto))
+                                        .characterEncoding("utf-8"))
+                        .andExpect(status().isOk())
+                        .andReturn();
+
+        courseOfferingDto =
+                objectMapper.readValue(
+                        mvcResult.getResponse().getContentAsString(), CourseOfferingDto.class);
+
+        CoopDto coopDto = new CoopDto();
+        coopDto.setStudent(studentDto);
+        coopDto.setStatus(CoopStatus.FUTURE);
+        coopDto.setCourseOffering(courseOfferingDto);
+
+        mvc.perform(
+                        post("/coops")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(objectMapper.writeValueAsString(coopDto))
+                                .characterEncoding("utf-8"))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        coopDto.setStudent(studentDto2);
+        coopDto.setStatus(CoopStatus.FUTURE);
+        coopDto.setCourseOffering(courseOfferingDto);
+
+        mvc.perform(
+                        post("/coops")
+                                .contentType(MediaType.APPLICATION_JSON)
+                                .content(objectMapper.writeValueAsString(coopDto))
+                                .characterEncoding("utf-8"))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        File testFile = new File("src/test/resources/sampleStudent.csv");
+        MultipartFile multipartFile =
+                new MockMultipartFile("Student CSV", new FileInputStream(testFile));
+
+        mvcResult =
+                mvc.perform(
+                                multipart("/csv-parser/check-registered")
+                                        .file("file", multipartFile.getBytes())
+                                        .param(
+                                                "course_id",
+                                                String.valueOf(courseOfferingDto.getId()))
+                                        .contentType(MediaType.MULTIPART_FORM_DATA)
+                                        .characterEncoding("utf-8"))
+                        .andExpect(status().isOk())
+                        .andReturn();
+
+        List<Object> students =
+                Arrays.asList(
+                        objectMapper.readValue(
+                                mvcResult.getResponse().getContentAsString(), Object[].class));
```

Comment:
You can change `Object[]` to `String[]` and change `List<Object>` to `List<String>`